ApiController.php

```diff
@@ -16,12 +16,14 @@
     public function doAction(CustomerReposiotry customer)
     {
 	// What kind of problem can we have when using global variables?
+//Concurrency issues//Testing and Confinement 
         $Case = $_GET["case"];
         $customer = $_POST["customer"];
         $product_id = $_GET["productId"];
 
         try {
 	// think about what could be done to avoid this logic in a controller
+	// try to move to services layer
```

Comment:
Exact, but how?

Reply:
in this case. i think we can apply factory method design patern and move to services layer .. not have logic in controller.

Reply:
That's a way, yes. Can you try to use it here?

Reply:
class Factory
{
    public function CreateProcess($type)
    {
        if ($type == 'product_id')
		{
            retrun new Product();
        } 
		if ($type == ''contact'')
		{
            retrun new Contact();
        } else{
            return new customer();
        }
    }
}
Factory method 
abstract Class MyProcess
{
    abstract function CreateProcess();

    public function Process()
    {
        $Product = $this->Create();
        ...
        ...
    }
}
///

class ProductProcess extends MyProcess
{
    function Create()
    {
        return new product();
    }
}
///
....

Reply:
Push a pull request please, in the git-flow way if possible

Reply:
i want to remind you .this is comment.

Reply:
What do you mean?

Reply:
i want to explain for you  .how to implement Factory method it here

Reply:
Yes, but can you actually code it?

Reply:
you can review again .this is actully code. .